tests/src/main/kotlin/se/ansman/kotshi/DefaultValues.kt

```diff
@@ -0,0 +1,125 @@
+package se.ansman.kotshi
+
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+
+@JsonSerializable
+data class ClassWithDefaultValues(
+        @UseJsonDefaultValue
+        val v1: WithCompanionFunction,
+        @UseJsonDefaultValue
+        val v2: WithStaticFunction,
+        @UseJsonDefaultValue
+        val v3: WithCompanionProperty,
+        @UseJsonDefaultValue
+        val v4: WithStaticProperty,
+        @UseJsonDefaultValue
+        val v5: GenericClassWithDefault<String>,
+        @UseJsonDefaultValue
+        val v6: GenericClassWithDefault<Int>,
+        @UseJsonDefaultValue
+        val v7: LocalDate,
+        @UseJsonDefaultValue
+        val v8: LocalTime,
+        @UseJsonDefaultValue
+        val v9: LocalDateTime,
+        val v10: WithCompanionFunction, // No annotations, should not get a default value
+        @DefaultQualifier
+        val v11: WithCompanionFunction,
+        @UseJsonDefaultValue
+        val v12: ClassWithConstructorAsDefault,
+        @UseJsonDefaultValue
+        val v13: GenericClassWithConstructorAsDefault<String>,
+        @UseJsonDefaultValue
+        val v14: Int
+)
+
+@JsonSerializable
+data class WithCompanionFunction(val v: String?) {
+    companion object {
+        @JsonDefaultValueProvider
+        fun provideDefault(): WithCompanionFunction = WithCompanionFunction("WithCompanionFunction")
+
+        @JsonDefaultValueProvider
+        @DefaultQualifier
+        fun provideQualifiedDefault() = WithCompanionFunction("DefaultQualifier")
+    }
+}
+
+@JsonSerializable
+data class WithStaticFunction(val v: String?) {
+    companion object {
+        @JsonDefaultValueProvider
+        @JvmStatic
+        fun provideDefault() = WithStaticFunction("WithStaticFunction")
+    }
+}
+
+@JsonSerializable
+data class WithCompanionProperty(val v: String?) {
+    companion object {
+        @get:JsonDefaultValueProvider
+        val defaultValue = WithCompanionProperty("WithCompanionProperty")
+    }
+}
+
+@JsonSerializable
+data class WithStaticProperty(val v: String?) {
+    companion object {
+        @JvmField
+        @field:JsonDefaultValueProvider
+        val defaultValue = WithStaticProperty("WithStaticProperty")
+    }
+}
+
+@JsonSerializable
+data class GenericClassWithDefault<out T>(val v: T?) {
+    companion object {
+        @JsonDefaultValueProvider
+        fun <T> provideDefault() = GenericClassWithDefault<T>(null)
+
+        @JsonDefaultValueProvider
+        fun provideIntDefault() = GenericClassWithDefault(4711)
+    }
+}
+
+object DefaultProvider {
+    @JsonDefaultValueProvider
+    fun provideDefaultLocalDate(): LocalDate = LocalDate.MIN
+
+    @JsonDefaultValueProvider
+    @JvmStatic
```

Comment:
Is it intended to mix JvmStatic and not? (I assume it is) If so, maybe add a comment to make it more obvious

Reply:
Both versions are tested.